packages/api/src/utils.js

```diff
@@ -33,23 +35,48 @@ const decodeStateTransition = async (client, base64) => {
 
   switch (decoded.type) {
     case StateTransitionEnum.DATA_CONTRACT_CREATE: {
+      const dataContractConfig = stateTransition.getDataContract().getConfig()
+
+      decoded.internalConfig = {
+        canBeDeleted: dataContractConfig.canBeDeleted,
+        readonly: dataContractConfig.readonly,
+        keepsHistory: dataContractConfig.keepsHistory,
+        documentsKeepHistoryContractDefault: dataContractConfig.documentsKeepHistoryContractDefault,
+        documentsMutableContractDefault: dataContractConfig.documentsMutableContractDefault,
+        documentsCanBeDeletedContractDefault: dataContractConfig.documentsCanBeDeletedContractDefault,
+        requiresIdentityDecryptionBoundedKey: dataContractConfig.requiresIdentityDecryptionBoundedKey ?? null,
+        requiresIdentityEncryptionBoundedKey: dataContractConfig.requiresIdentityEncryptionBoundedKey ?? null
+      }
+
       decoded.dataContractId = stateTransition.getDataContract().getId().toString()
-      decoded.identityId = stateTransition.getOwnerId().toString()
+      decoded.ownerId = stateTransition.getOwnerId().toString()
+      decoded.schema = stateTransition.getDataContract().getDocumentSchemas()
+      decoded.signature = Buffer.from(stateTransition.toObject().signature).toString('hex')
+      decoded.signatureKeyId = stateTransition.toObject().signaturePublicKeyId
+      decoded.raw = stateTransition.toBuffer().toString('hex')
 
       break
     }
     case StateTransitionEnum.DOCUMENTS_BATCH: {
       decoded.transitions = stateTransition.getTransitions().map((documentTransition) => ({
         id: documentTransition.getId().toString(),
```

Comment:
Different document types have different field set

Here is what is missing:
* Create - data, prefundedVotingBalance, identityContractNonce, documentType
* Replace - data, identityContractNonce, documentType
* Transfer - identityContractNonce, documentType
* UpdatePrice - price, documentType, identityContractNonce
* Purchase - price, identityContractNonce, documentType

Reply:
document type already exist

Reply:
✅